pkg/telemetry/opentelemetry.go

```diff
@@ -0,0 +1,135 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/goto/salt/log"
+	"go.opentelemetry.io/contrib/instrumentation/host"
+	"go.opentelemetry.io/contrib/instrumentation/runtime"
+	"go.opentelemetry.io/contrib/samplers/probability/consistent"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"google.golang.org/grpc/encoding/gzip"
+)
+
+type OpenTelemetryConfig struct {
+	Enabled                bool          `yaml:"enabled" mapstructure:"enabled" default:"true"`
+	CollectorAddr          string        `yaml:"collector_addr" mapstructure:"collector_addr" default:"localhost:4317"`
+	PeriodicReadInterval   time.Duration `yaml:"periodic_read_interval" mapstructure:"periodic_read_interval" default:"1s"`
+	TraceSampleProbability float64       `yaml:"trace_sample_probability" mapstructure:"trace_sample_probability" default:"1"`
+}
+
+func initOTLP(ctx context.Context, cfg Config, logger log.Logger) (func(), error) {
+	if !cfg.OpenTelemetry.Enabled {
+		return noOp, nil
+	}
```

Comment:
Let's add log when it's disabled similar to newrelic one
```suggestion
	if !cfg.OpenTelemetry.Enabled {
		logger.Info("OpenTelemetry monitoring is disabled.")
		return noOp, nil
	}
```